src/sources/opentelemetry/tests.rs

```diff
@@ -42,7 +47,156 @@ async fn receive_grpc_logs() {
+            .await
+            .unwrap();
+        tokio::spawn(server);
+        test_util::wait_for_tcp(grpc_addr).await;
+
+        // send request via grpc client
+        let mut client = LogsServiceClient::connect(format!("http://{}", grpc_addr))
+            .await
+            .unwrap();
+        let req = Request::new(ExportLogsServiceRequest {
+            resource_logs: vec![ResourceLogs {
+                resource: Some(OtelResource {
+                    attributes: vec![KeyValue {
+                        key: "res_key".into(),
+                        value: Some(AnyValue {
+                            value: Some(any_value::Value::StringValue("res_val".into())),
+                        }),
+                    }],
+                    dropped_attributes_count: 0,
+                }),
+                scope_logs: vec![ScopeLogs {
+                    scope: None,
+                    log_records: vec![LogRecord {
+                        time_unix_nano: 1,
+                        observed_time_unix_nano: 2,
+                        severity_number: 9,
+                        severity_text: "info".into(),
+                        body: Some(AnyValue {
+                            value: Some(any_value::Value::StringValue("log body".into())),
+                        }),
+                        attributes: vec![KeyValue {
+                            key: "attr_key".into(),
+                            value: Some(AnyValue {
+                                value: Some(any_value::Value::StringValue("attr_val".into())),
+                            }),
+                        }],
+                        dropped_attributes_count: 3,
+                        flags: 4,
+                        // opentelemetry sdk will hex::decode the given trace_id and span_id
+                        trace_id: str_into_hex_bytes("4ac52aadf321c2e531db005df08792f5"),
+                        span_id: str_into_hex_bytes("0b9e4bda2a55530d"),
+                    }],
+                    schema_url: "v1".into(),
+                }],
+                schema_url: "v1".into(),
+            }],
+        });
+        let _ = client.export(req).await;
+        let mut output = test_util::collect_ready(logs_output).await;
+        // we just send one, so only one output
+        assert_eq!(output.len(), 1);
+        let event = output.pop().unwrap();
+        schema_definition.assert_valid_for_event(&event);
+
+        assert_eq!(event.as_log().get(".").unwrap(), &vrl::value!("log body"));
+
+        let meta = event.as_log().metadata().value();
+        assert_eq!(
+            meta.get(path!("vector", "source_type")).unwrap(),
+            &vrl::value!(OpentelemetryConfig::NAME)
+        );
+        assert!(meta
+            .get(path!("vector", "ingest_timestamp"))
+            .unwrap()
+            .is_timestamp());
+        assert_eq!(
+            meta.get(path!("opentelemetry", "resources")).unwrap(),
```

Comment:
nit: can use `OpentelemetryConfig::NAME` for future proofing (I failed to do this on at least on source)

Reply:
I'm of the opinion that you shouldn't in tests. Then if the constant is changed by mistake the tests will fail and catch that change. If we mean to change it the test fail and will need updating intentionally.

Reply:
Hmm interesting. I can see the advantage of detecting unintended changes to the constant via unit tests. But I wonder if that would maybe a happy compromise might be something like a single unit test which validates all the named components against the expected string literal? As opposed to using the string literal in multiple test cases throughout potentially multiple locations in the code. Seems like something that would go in the named component unit testing ? / cc @tobz